lib/pages/profile_info/profile_info_body/profile_info_body.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:async';
+
+import 'package:dartz/dartz.dart' hide State;
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/contact/lookup_match_contact_state.dart';
+import 'package:fluffychat/domain/usecase/contacts/lookup_match_contact_interactor.dart';
+import 'package:fluffychat/pages/profile_info/profile_info_body/profile_info_body_view.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+import 'package:matrix/matrix.dart';
+
+class ProfileInfoBody extends StatefulWidget {
+  const ProfileInfoBody({
+    required this.user,
+    required this.parentContext,
+    this.onNewChatOpen,
+    Key? key,
+  }) : super(key: key);
+
+  final User? user;
+
+  final BuildContext parentContext;
+
+  final void Function()? onNewChatOpen;
+
+  @override
+  State<ProfileInfoBody> createState() => ProfileInfoBodyController();
+}
+
+class ProfileInfoBodyController extends State<ProfileInfoBody> {
+  final _lookupMatchContactInteractor =
+      getIt.get<LookupMatchContactInteractor>();
+
+  StreamSubscription? lookupContactNotifierSub;
+
+  final ValueNotifier<Either<Failure, Success>> lookupContactNotifier =
+      ValueNotifier<Either<Failure, Success>>(
+    const Right(LookupContactsInitial()),
+  );
+
+  User? get user => widget.user;
+
+  BuildContext get parentContext => widget.parentContext;
+
+  bool get isOwnProfile => user?.id == user?.room.client.userID;
+
+  void lookupMatchContactAction() {
+    if (user == null) return;
+    lookupContactNotifierSub = _lookupMatchContactInteractor
+        .execute(
+          val: user!.id,
+        )
+        .listen(
+          (event) => lookupContactNotifier.value = event,
+        );
+  }
+
+  void openNewChat(String roomId) {
+    parentContext.go('/rooms/$roomId');
```

Comment:
please demo in the cases:
- with person we have chat before
- with person we have not had

Reply:
you should go to `draftChat` in the case no existed chat with this person before

Reply:
[Capture vidéo du 19-03-2024 11:38:37.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/f0fdcc1b-52ae-4499-8f96-aeecbea0fd02)

[Capture vidéo du 19-03-2024 11:48:01.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/f69aa606-ce51-4fb8-82e9-49e0054cf6be)

Reply:
@hoangdat fixed: https://github.com/linagora/twake-on-matrix/pull/1596/commits/d6b1297ccafbafb4cc7563e972b5a28a8f96c40b

[Capture vidéo du 21-03-2024 11:33:47.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/85f3b275-50d3-4afb-bb50-7bfd2eac1a96)